crates/lib/src/execution_data.rs

```diff
@@ -228,6 +225,11 @@ impl ReadoutMap {
     }
 }
 
+struct MemoryReference {
+    name: String,
+    index: usize,
+}
+
```

Comment:
I'm wary of a temporary solution of creating a new `MemoryReference` type. I appreciate that this is module private, at least. If/when we find a better solution for the `u64`/`usize` debate, I'd like to see this removed in favor of the `quil-rs` one again.

Which is a long way to say... is there an issue for replacing this in the future?

Reply:
Yeah, I don't love it and would like to rely on `quil-rs` for this. That said I don't think it's _that_ bad. At no point does any of this code flow through `quil-rs`, we're really just parsing the strings we get back from controller service into a structure that works for building a `ReadmoutMap`. Would be nice to not have to duplicate the parsing logic, but there is a separation of concerns and no communication with `quil-rs` that might break on account of the structs drifting.

Regardless, I created #224 and added a comment.